client/my-sites/plugins-wpcom/plugin-install-button.jsx

```diff
@@ -3,22 +3,78 @@
  */
 import React from 'react';
 import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+import { get } from 'lodash';
+import page from 'page';
 
 /**
  * Internal dependencies
  */
 import Button from 'components/button';
+import { getSelectedSite } from 'state/ui/selectors';
+import { getEligibility } from 'state/automated-transfer/selectors';
+import { initiateThemeTransfer } from 'state/themes/actions';
+import { setAutomatedTransferStatus } from 'state/automated-transfer/actions';
 
-export const WpcomPluginInstallButton = ( { translate, disabled, plugin, site } ) => {
-	return <Button
-		onClick={ undefined }
-		primary={ true }
-		type="submit"
-		disabled={ disabled }
-		href={ `/plugins/${ plugin.slug }/${ site.slug }/eligibility` }
-	>
-		{ translate( 'Install' ) }
-	</Button>;
+export const WpcomPluginInstallButton = props => {
+	const {
+		translate,
+		disabled,
+		plugin,
+		siteSlug,
+		siteId,
+		eligibilityData,
+		navigateTo,
+		initiateTransfer,
+		setStatus
+	} = props;
+
+	function installButtonAction( event ) {
+		event.preventDefault();
+
+		const hasErrors = !! get( eligibilityData, 'eligibilityHolds', [] ).length;
+		const hasWarnings = !! get( eligibilityData, 'eligibilityWarnings', [] ).length;
+
+		if ( ! hasErrors && ! hasWarnings ) {
+			// No need to show eligibility warnings page, initiate transfer immediately
+			// TODO: refactor to use plugin slug
+			initiateTransfer( siteId, null );
+			setStatus( siteId, 'start' );
```

Comment:
is this `setStatus()` call temporary? Looks like the `start` status will be set via the actions in [this PR](https://github.com/Automattic/wp-calypso/pull/10472/files#diff-2ea520ab23f0b8cf94522bab515c5d60R26).